### DISCUSSION TITLE
0.11.0 Release

### DISCUSSION THREAD

**Discussion by blakeblackshear** [blakeblackshear/frigate#3887] [General]
# Major Changes for 0.11.0

### 🚨 🚨 BREAKING CHANGES 🚨 🚨 
- Update to ffmpeg5 will break any custom input arguments using the `stimeout` parameter. This needs to be renamed to `timeout` in your config.
- Some amd64 users may need to set the environment variable `LIBVA_DRIVER_NAME=i965` to get similar hardware acceleration performance to previous versions.
- Upgrading will run migrations on the database file that are not backwards compatible. In order to rollback to 0.10.0, you will either need to restore an old database file or delete the database files so they can be recreated.
- This version includes a significant rework of the docker build. This may have broken ffmpeg hardware acceleration for some hardware that was working previously. Please open an issue if this happens so we can try and track down what's missing.
- Some deployment approaches are overwriting the PATH environment value to an old value. If you see errors like `No such file or directory: 'ffmpeg'`, then you need to stop overriding your PATH. This happens with both Proxmox LXC and Portainer (#3223)
- The `max_seconds` config setting under events has been removed as it no longer has any effect

## Frigate+ Integration

With this release, users can easily send snapshots to [Frigate+](https://frigate.video/plus/) for labeling directly from the Events view. Note that Frigate+ is still in development, and custom models are not yet available. There is no cost to create an account and submit labeled images. Making it simpler to submit images should accelerate the build out of a robust training set.

To enable the Frigate+ functionality, you need to obtain an API Key from your Frigate+ account on the [settings page](https://plus.frigate.video/dashboard/settings/) and set it via the `PLUS_API_KEY` environment variable or in the addon configuration page. **Note that this cannot be set in the yml configuration file. It must be set in your compose file or docker command.**

![image](https://user-images.githubusercontent.com/569905/168949909-3032fbcf-8014-45c5-86cc-e1d8ff7ca41d.png)

Here is the roadmap for Frigate+:
- [x] Launch web version allowing manual upload of training examples and labeling
- [x] Integrate submission in Frigate
- [ ] Launch Frigate optimized models
- [ ] Launch individually optimized models
- [ ] Integrate labeling and suggestions in Frigate

## Updated Events View

- Speed. The events view has been completely revamped and should be much faster than it was before.
- Slightly larger thumbnails.
- You no longer have to refresh the page to see the latest events.
- You can "star" or "favorite" events to tell Frigate to keep associated recordings and the snapshot indefinitely.
- Sub labels have been added to allow setting arbitrary labels on events. For example, you can use deepstack to run facial recognition and update the event with the name of the person detected. (#2949) 

## Updated Recordings View

- Speed. The recordings view doesn't look different, but it has been completely overhauled. It should be much faster, and event data and thumbnails are lazy loaded to reduce the amount of data that is loaded when visiting the page.
- Smoother seeking and transitions. Re-renders of the player have been reduced and you should see smoother seek behavior when selecting events.
- 16x playback was added for browsers other than Firefox.

## Updated Dockerbuild

- Frigate is now built on top of plain Debian containers and consolidated into a single multiarch image. Support for nvidia decoding is built into the single image. There is no longer a need for a special tag.

## Experimental Timeline UI
Not recommended for use yet, but an alternative timeline UI has been added. It has very limited browser/device support. #2830 

## Other changes
- MQTT sensors for any object type #2908 
- Update latest object endpoints to reference most recent event from database (you will no longer see black images for objects under the camera in live view) #2944 
- Added a new ratio filter for object filters #2952
- Camera recordings always in sidebar #2846
- Enable and set mode per camera for birdseye #3083 
- Customize order and hide cameras from the UI #2981 
- Enable improve_contrast via MQTT #3011 
- Enable/disable motion detection independent of object detection via MQTT #3117 
- MQTT topic for motion detection #3152 
- Batch deletes of recordings when more than 100,000 recordings need to be deleted
- Switch to Btbn ffmpeg builds
- Improve cache handling for thumbnails and snapshots of completed events
- Fixes hwaccel issues for Raspberry Pi users 
- Updates to debian 11 as a base image
- Consolidate Dockerfile to enable multiarch images
- Birdseye icon can now be customized #2979
- Frigate now checks for a newer release on startup and exposes that in the stats endpoint #3038 
- Initial implementation of mypy and type checking
- Images from backend now have appropriate cache headers
- Motion `contour_area` and `threshold` values can now be adjusted dynamically via mqtt (#3154)
- Fix bug with bounding boxes outside of frame
- Ensure playlist is in front of error messages in videojs
- Fixes for H265 events and recordings in browsers with native support for H265 (thanks @uvjustin)
- Fixes for stuttering playback issues in some circumstances (thanks @uvjustin)

**Full Changelog**: https://github.com/blakeblackshear/frigate/compare/v0.10.1...v0.11.0

Image: `blakeblackshear/frigate:0.11.0`

<hr /><em>This discussion was created from the release <a href='https://github.com/blakeblackshear/frigate/releases/tag/v0.11.0'>0.11.0 Release</a>.</em>

**Comment by sheminasalam** [comment]
Hi there, I am getting the following error on my rpi4 running HA core installation.
```
[supervisor.docker.addon] Invalid build environment, can't build this add-on!
```

Solved: The addon needs to be uninstalled/reinstalled if you are seeing this error.

**Reply by NickM-27** [reply]
I think you mean HA supervised installation since your logs have supervisor?

Also, what OS are you running?

**Reply by sheminasalam** [reply]
raspbian os

**Reply by cvroque** [reply]
I get the same error, running HAOS on a NUC.

`[supervisor.docker.addon] Invalid build environment, can't build this add-on!`

**Reply by b3nj1** [reply]
In case anyone else hits this. When building on Debian bullseye amd64, `make build_web` fails without any logging. I've gotten around it by changing the node docker image and setting the npm cache to /tmp.

```
docker run -e npm_config_cache=/tmp/npmcache --volume ${PWD}/web:/web -w /web --volume /etc/passwd:/etc/passwd:ro --volume /etc/group:/etc/group:ro -u $(CURRENT_UID):$(CURRENT_GID) node:current-bullseye /bin/bash -c "npm install && npm run build"
```

**Reply by NickM-27** [reply]
@b3nj1 ~are you also running in supervised?~

**Reply by blakeblackshear** [reply]
I was able to install the addon without any issues on my rpi4 running HAOS. Have you tried uninstalling and reinstalling instead of upgrading?

**Reply by blakeblackshear** [reply]
The addon pulls the prebuilt container. It does not build anything itself.

**Reply by niemyjski** [reply]
@blakeblackshear Any idea why it's failing then, really does seem weird it's building anything,

**Reply by blakeblackshear** [reply]
The only think I can think of is the change in versioning strategy. This went from 3.1 -> 0.11.0, so maybe its because the version appears to be backwards and it only impacts users upgrading. I still haven't been able to reproduce.

**Reply by cvroque** [reply]
> I was able to install the addon without any issues on my rpi4 running HAOS. Have you tried uninstalling and reinstalling instead of upgrading?

While updating failed, uninstalling and installing again worked! 
In case it is useful I attached the supervisor logs.
[frigateupdate.txt](https://github.com/blakeblackshear/frigate/files/9630393/frigateupdate.txt)

**Reply by sheminasalam** [reply]
> > I was able to install the addon without any issues on my rpi4 running HAOS. Have you tried uninstalling and reinstalling instead of upgrading?
> 
> While updating failed, uninstalling and installing again worked! In case it is useful I attached the supervisor logs. [frigateupdate.txt](https://github.com/blakeblackshear/frigate/files/9630393/frigateupdate.txt)

this worked

**Comment by craigmate** [comment]
HI

I am running Frigate in UNRAID, but its not showing any update.  I have uninstalled and re-installed, but my version is still 0.10.1-83481af
Any ideas where I might be going wrong.

Thank You

**Reply by yayitazale** [reply]
Unraid template has been updated with [this PR](https://github.com/yayitazale/unraid-templates/pull/5), so you will need to uninstall the actual APP and install the new one with the new unified template from CA store ([as I mentioned in the support forum](https://forums.unraid.net/topic/98064-support-blakeblackshear-frigate/))

Anyway, CA store will take some time to show the new app

**Reply by yayitazale** [reply]
The change is now reflected, so you can install the v11.0 from the CA and delete the old APP

**Comment by sym0nd0** [comment]
Following the update, I have a Frigate motion sensor that is going wild. Detecting and then clearing detection every 31 second since updating yesterday, but there's nothing/no-one there. 

Debug with motion boxes enables doesn't pick up anything. 😕

Anyone else?

![Screenshot_20220924-063359_Home Assistant.jpg](https://user-images.githubusercontent.com/55590428/192081992-c4aa63e4-dbd6-43c4-9122-c1148795eb03.jpg)

For now, I've just disabled the entity so it doesn't make MQTT or HA fall over.

**Reply by NickM-27** [reply]
You can customize the amount of time before it resets to no motion. But even with that MQTT can handle it no problem. 

It is just general motion, so any single motion box will trigger it.

**Comment by DanGarion** [comment]
Wow... night and day difference on the speed of Frigate pulling up live video... It used to be 5-10 seconds to get the live video to display and stop "buffering" but now it is nearly instantaneous. What a great update!

**Comment by p51d8102** [comment]
Probably a newb question but my docker-compose.yaml file pulls the image blakeblackshear/frigate:stable-amd64.  Should I work on changing this or will the 0.11.0 update get the stable flag anytime soon?

**Reply by NickM-27** [reply]
It's stable. We use multiarch now so just remove the `amd64`

**Comment by p51d8102** [comment]
That worked. Thanks Nicolas!

**Comment by grembling22** [comment]
Since upgrading to version 11 I am no longer getting images sent via telegram. When I browse to /api/<camera name>/person/best.jpg no images are loading. If I switch back to version 10 it works again. Looked through the release notes and didn't see anything mentioned.

**Reply by NickM-27** [reply]
The best API was left in for compatibility but its behavior has changed. See https://github.com/blakeblackshear/frigate/blob/master/frigate/http.py#L353

**Reply by grembling22** [reply]
I don't quite understand that. What would be the best method to send images to telegram now?

**Reply by NickM-27** [reply]
What kind of image are you sending? (Like under what circumstance?)

I'm surprised you were using the `best` to begin with since that doesn't always change for each event and could be the same image quite often.

**Reply by grembling22** [reply]
it's an automation that sends the best.jpg on a person detection. If I remember correctly, this was how the frigate docs recommended it be setup. Granted this was a few versions ago.

**Reply by NickM-27** [reply]
Yeah I think the best way now would be to just use `/<camera_name>/<label>/thumbnail.jpg`

**Reply by grembling22** [reply]
I tried that method as well but only get a black image.

**Reply by NickM-27** [reply]
Do you have events enabled? This pulls the thumbnail from the latest event

**Reply by grembling22** [reply]
I do.

**Reply by NickM-27** [reply]
What url did you try? If there is an event with that camera (which has snapshots enabled) and label then it will return a thumbnail

**Reply by haggercody** [reply]
hmm I think I hit this too?  I use the HA api, though which is http:/host:8123/api/frigate/notifications/{event_id}/thumbnail.jpg
Trying this with a valid event id i am not seeing anything.  I did check the link to a thumbnail of an image in the frigate UI and the url was like this:
http://host:8123/api/hassio_ingress/{some key value}/api/events/{event_id}/thumbnail.jpg

**Reply by haggercody** [reply]
I got mine to come back by doing a relative path, "/api/frigate/events/{event_id}/thumbnail.jpg"
noticed the notifications part changed to events?

**Reply by haggercody** [reply]
well shoot. I took a HA update and now my image paths are invalid again, 
for example with telegarm:
`Can't send file with kwargs: {'caption': 'A person was detected at front door.', 'url': '/api/frigate/notifications/1670246047.182908-xxxcx2/thumbnail.jpg?format=ios'}`

Oddly if I set the full path again, i get the same error.

**Comment by bm55** [comment]
@blakeblackshear thanks for adding the option to view snapshot! It’s very helpful to update configuration to remove unwanted/false notifications.

**Reply by NickM-27** [reply]
Thank @mellbye for that one 👍

**Reply by bm55** [reply]
Thanks @mellbye !

**Comment by Bugeyedearl** [comment]
Currently frigate working on .v10 but when upgrading to v.11.1 i'm getting error ffmpeg process crash. Anyone knows how correct this i'm fairly new to frigate?

[2022-10-14 08:55:17] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:55834]
[2022-10-14 08:55:18] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:8082 | Remote => 127.0.0.1:47298]
[2022-10-14 08:55:22] watchdog.front                 INFO    : No frames received from front in 20 seconds. Exiting ffmpeg...
[2022-10-14 08:55:22] watchdog.front                 INFO    : Waiting for ffmpeg to exit gracefully...
[2022-10-14 08:55:22] frigate.video                  ERROR   : front: Unable to read frames from ffmpeg process.
[2022-10-14 08:55:22] frigate.video                  ERROR   : front: ffmpeg process is not running. exiting capture thread...
[2022-10-14 08:55:32] watchdog.front                 ERROR   : Ffmpeg process crashed unexpectedly for front.
[2022-10-14 08:55:32] watchdog.front                 ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-10-14 08:55:32] ffmpeg.front.detect            ERROR   : [rtsp @ 0x55f53d0a0dc0] Could not find codec parameters for stream 0 (Video: h264, none): unspecified size
[2022-10-14 08:55:32] ffmpeg.front.detect            ERROR   : Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[2022-10-14 08:55:32] ffmpeg.front.detect            ERROR   : Output file #0 does not contain any stream
[2022-10-14 08:55:52] watchdog.front                 INFO    : No frames received from front in 20 seconds. Exiting ffmpeg...
[2022-10-14 08:55:52] watchdog.front                 INFO    : Waiting for ffmpeg to exit gracefully...
[2022-10-14 08:55:52] frigate.video                  ERROR   : front: Unable to read frames from ffmpeg process.
[2022-10-14 08:55:52] frigate.video                  ERROR   : front: ffmpeg process is not running. exiting capture thread...

**Reply by NickM-27** [reply]
Please make a new issue, there is not enough info here to help you

**Comment by pcsmith811** [comment]
Is there anything beside's entering the API key (PLUS_API_KEY) in a docker compose to be able to send images for training?:

    environment:
      FRIGATE_RTSP_PASSWORD: "password"
      LIBVA_DRIVER_NAME: i965
      PLUS_API_KEY: "REDACTED"

I've done this and the API key shows up under the running container like it should but I don't have a "Send to Frigate+" button despite that and configuring all the same cameras with the same names from the config.yaml in the Frigate+ portal.   Thanks!

**Reply by NickM-27** [reply]
Please do not share your api key. Also, do be aware that snapshots must be enabled to be able to submit to Frigate+

**Reply by DanGarion** [reply]
With that said how do we get it to work? I just get this message every time.
![image](https://user-images.githubusercontent.com/1666375/198152919-4d4af8c5-3c18-49b7-93e1-b29879dd7ae0.png)

**Reply by pcsmith811** [reply]
> Please do not share your api key. Also, do be aware that snapshots must be enabled to be able to submit to Frigate+

Ahh snapshots! - that was the key.   Also, thanks for redacting, but if you had looked closely part of it said notreal and I changed a bunch of the key :)   I wanted to make sure I also had it in the right format.

**Reply by NickM-27** [reply]
@DanGarion if it isn't working then you haven't entered the env key correctly. If you are running in the addon then be aware you MUST enter the key via the config option in the HA UI and not using the frigate.yaml as that is not supported

**Reply by DanGarion** [reply]
Hah! I didn't know that was where you entered it until you mentioned that and it clicked! :)
![image](https://user-images.githubusercontent.com/1666375/198155074-81a08c0c-e2c9-4e34-8b5d-132a3585898d.png)
